src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -312,6 +324,44 @@ public AnalyzerConfigOptions GetOptionsForSourcePath(string path)
                 => new DictionaryAnalyzerConfigOptions(_projectState._lazyAnalyzerConfigOptions.GetValue(CancellationToken.None).GetOptionsForSourcePath(path).AnalyzerOptions);
         }
 
+        /// <summary>
+        /// Provides editorconfig options for Razor design-time documents.
+        /// Razor does not support editorconfig options but has custom settings for a few formatting options whose values
+        /// are only available in-proc and the same for all Razor design-time documents.
+        /// This type emulates these options as analyzer config options.
+        /// </summary>
+        private sealed class RazorDesignTimeAnalyzerConfigOptions : AnalyzerConfigOptions
+        {
+            private readonly ILegacyGlobalOptionsWorkspaceService? _globalOptions;
+
+            public RazorDesignTimeAnalyzerConfigOptions(HostWorkspaceServices services)
+            {
+                // not available OOP:
```

Comment:
Is that a problem here?

Reply:
Yes, but not bigger than it already has been. OOP features do not work with Razor today.